src/Microsoft.NetCore.Analyzers/Core/Security/UseSharedAccessProtocolHttpsOnly.cs

```diff
@@ -0,0 +1,140 @@
+
+            // Security analyzer - analyze and report diagnostics on generated code.
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze | GeneratedCodeAnalysisFlags.ReportDiagnostics);
+
+            context.RegisterCompilationStartAction(compilationStartAnalysisContext =>
+            {
+                var microsoftWindowsAzureStorageNamespaceSymbol = compilationStartAnalysisContext
+                            .Compilation
+                            .GlobalNamespace
+                            .GetMembers("Microsoft")
+                            ?.FirstOrDefault()
+                            .GetMembers("WindowsAzure")
+                            .OfType<INamespaceOrTypeSymbol>()
+                            .FirstOrDefault()
+                            .GetMembers("Storage")
+                            .OfType<INamespaceOrTypeSymbol>()
+                            .FirstOrDefault();
+
+                if (microsoftWindowsAzureStorageNamespaceSymbol == null)
+                {
+                    return;
+                }
+
+                var wellKnownTypeProvider = WellKnownTypeProvider.GetOrCreate(compilationStartAnalysisContext.Compilation);
+
+                compilationStartAnalysisContext.RegisterOperationBlockStartAction(operationBlockStartContext =>
+                {
+                    var owningSymbol = operationBlockStartContext.OwningSymbol;
+                    var interproceduralAnalysisConfig = InterproceduralAnalysisConfiguration.Create(
+                                                            operationBlockStartContext.Options,
+                                                            SupportedDiagnostics,
+                                                            defaultInterproceduralAnalysisKind: InterproceduralAnalysisKind.None,
+                                                            cancellationToken: operationBlockStartContext.CancellationToken,
+                                                            defaultMaxInterproceduralMethodCallChain: 1);
+
+                    operationBlockStartContext.RegisterOperationAction(operationAnalysisContext =>
+                    {
+                        var invocationOperation = (IInvocationOperation)operationAnalysisContext.Operation;
+                        var methodSymbol = invocationOperation.TargetMethod;
+
+                        if (methodSymbol.Name != "GetSharedAccessSignature")
+                        {
+                            return;
+                        }
+
+                        var namespaceSymbol = methodSymbol.ContainingNamespace;
+
+                        if (namespaceSymbol == null)
+                        {
+                            return;
+                        }
+
+                        var namespaceQualifiedName = namespaceSymbol.ToDisplayString();
+
+                        if (namespaceQualifiedName != "Microsoft.WindowsAzure.Storage" &&
```

Comment:
Instead of using string comparisons, I would recommend walking up the ContainingNamespace chain until you reach a namespace that Equals `microsoftWindowsAzureStorageNamespaceSymbol`.